terraform/variables.tf

```diff
@@ -113,35 +105,70 @@ variable "k8s_nodepool_container_runtime" {
   }
 }
 
-variable "k8s_nodepool_flavor" {
+# `default` nodepool settings
+
+variable "k8s_default_nodepool_flavor" {
   type = map(string)
   description = "Flavor name of the instances that will be created for the node pool"
 
+  # You can get the list of available flavors here:
+  # https://www.scaleway.com/en/pricing/
+
   default = {
     production = "GP1-XS"
+    preprod = "DEV1-M"
+  }
+}
+
+variable "k8s_default_nodepool_size" {
+  type = map(number)
+  description = "Number of nodes desired in the default pool"
+
+  default = {
+  }
```

Comment:
I noticed something inconsistent: for the above variable you define the default here whereas for other variables you define the default in the terraform definition file. Maybe define all defaults here?

Reply:
I agree with you, I also find it inconsistent.
But I have mixed feelings about this, due to the facts that all variables are maps with the TF workspace name as key:

- Defining a default value in the  `lookup()` call allows to define a default value whatever the workspace name is.

- Defining all the default values in `variables.tf`, as you suggested, seems more logic. But by doing it there, we would assume that everyone is using a specific list of workspaces names (like `default`, `production` or `preprod`) and we would not provide a default value for other workspaces names.

WDYT?

Reply:
interesting. Technically we could have a `default` option in the default mappings? I leave it up to you to decide if it is better than what we have now :man_shrugging: